cmd/virt-probe/virt-probe.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017, 2018 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	goflag "flag"
+	"fmt"
+	"os"
+	"runtime/pprof"
+
+	"github.com/spf13/pflag"
+
+	"kubevirt.io/client-go/log"
+	cmdclient "kubevirt.io/kubevirt/pkg/virt-handler/cmd-client"
+)
+
+func main() {
+	// set new default verbosity, was set to 0 by glog
+	goflag.Set("v", "2")
+
+	socket := pflag.String("socket", cmdclient.SocketOnGuest(), "Socket for connecting to the cmd server")
+	domainName := pflag.String("domainName", "", "Domain Name of the Virtual Machine to connect the agent to. Usually namespace_vmname")
+	command := pflag.String("command", "", "Command to execute on the guest")
+	memProfile := pflag.String("memProfile", "", "Path to store a memory profile. Profiling is skipped if empty")
+
+	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.Parse()
+
+	log.InitializeLogging("virt-probe")
+
+	client, err := cmdclient.NewClient(*socket)
```

Comment:
what is the advantage of having virt-probe talk to virt-launcher which then talks connectso to the guest agent? Could we not get the same result wiht having virt-probe execute this command with the client itself?

Reply:
It hands of the logic on how to talk to the guest directly (which socket etc.) to virt-launcher.
I didn't want to duplicate the logic and also wanted to keep virt-probe as small as possible.
Also we thought about exposing the exec functionality as an API at some point so it would help to already have it in the api.